### PR TITLE
Workflow cicd pipeline

### PR DESCRIPTION
**Pull request by Hermann27**
This PR adds a full CI/CD pipeline for the project, which includes the following: 

- **Node.js and npm setup**: Installs Node.js and dependencies. 
- **Test workflow**: Runs tests whenever code is pushed to 'main' or on pull requests. 
- **Deployment workflow**: Deploys the project to xyz after successful tests.

### CODE REVIEW

**Comment by robstrader** [comment]
@Hermann27 the actions look good, but @ozzydogandev has now merged in his changes for the homepage, so now the example test config is broken. You will need to update this branch with the latest, then fix the issue. The error is that jest does not know how to handle .css files. Please see the following section of the docs: https://jestjs.io/docs/webpack#handling-static-assets

You may even want to fix that in a separate pull request, then update this PR after that one has been merged. Or you can fix it in this one.

**Comment by robstrader** [comment]
To update your PR branch, you have 2 choices, you can rebase or merge locally, then push to the remote branch, or you can update the branch remotely. Updating remotely is fairly simple when there are no merge conflicts, you can simply click the 'Update branch' button in the PR. Then do a hard reset on your local branch to the remote branch:

update-branch-pr

After clicking 'Update branch':

```sh
git checkout pr-mybranch
git fetch
git reset --hard origin/pr-mybranch
```

**Comment by Hermann27** [comment]
PR updated 👍

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build",
-    "lint": "eslint . --report-unused-disable-directives --max-warnings 0",
+    "lint": "eslint . --fix --report-unused-disable-directives --max-warnings 0",
```
Comment (robstrader) [inline package.json]: I'm not sure you actually want to use `--fix` here. You would like `lint` to report an errors. You could create a separate script for fixing named `lint:fix`.

File: jest.config.mjs [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
   testEnvironment: "jest-environment-jsdom",
-  transform: {
+  /* transform: {
     "^.+\\.jsx?$": "babel-jest",
   },
-  moduleNameMapper: {
+ moduleNameMapper: {
     "^@/(.*)$": "<rootDir>/src/$1",
-  },
+  },*/
```
Comment (robstrader) [inline jest.config.mjs]: Could you go ahead and remove the commented out lines.

**Review by robstrader — COMMENTED** [review]
Overall, looks great! Added a couple of comments. Good job on getting Jest working with CSS files.

File: jest.config.mjs (hunk 2 above)
Comment (Hermann27) [reply jest.config.mjs]: I have removed the comment as requested.

File: package.json (hunk 1 above)
Comment (Hermann27) [reply package.json]: I have added a separate script for fixing the lint.

**Review by robstrader — APPROVED** [review]
Sorry, meant to get back to this. Looks good to me, but you are going to need to merge in the latest changes from main as Ozzy has been pushing some changes.

**Comment by Hermann27** [comment]
image

I have updated this PR with the latest changes as Ozzy has pushed some changes. Now I need to fix all those eslint error

**Comment by Hermann27** [comment]
for now, I have turned off these rules:

- "react/prop-types": "off",

- "react/react-in-jsx-scope": "off",

- "react/no-unescaped-entities": "off",

What would be the best way to correct those errors?

**Pull request merged**